src/lib.rs

```diff
@@ -134,6 +134,7 @@ extern crate alloc;
 pub mod inclusive_map;
 pub mod inclusive_set;
 pub mod map;
+pub mod operations;
```

Comment:
Is this only public because it contains `Union` and `Intersection` iterators? I think I'd prefer to keep `operations` private, and wrap those very generic iterator structs up in a simpler, more purpose-specific struct for the public API.

I.e. instead of this in the public API:

```
pub struct Union<'a, T, L, R = L> { // ...
```

Have separate...

```
pub struct Union<T> { // ...
```

for each kind of set.

(Then we can separately think about whether the more flexible version is useful to have in the public API.)

Reply:
Makes sense — I had them public at first because they are generic over the iterator type and could thus be used with other iterators that emit ranges. However, I agree that it makes sense to keep things simple first and only export the types we are actually using.